proto/backup.proto

```diff
@@ -92,12 +92,23 @@ message ClusterIDError {
     uint64 request = 2;
 }
 
+// Should we let this be an generic error?
+message ExternalStorageError {
+    bool retryable = 1;
+
+    // If use_backoff_time is true, 
+    // the client SHOULD back off for (backoff_time) millseconds.
+    bool use_backoff_time = 2;
+    uint32 backoff_time = 3;
```

Comment:
what the client should use if `retryable == true && use_backoff_time == false`?

it seems like the whole message should have one field: `uint32 retry_after`. 
* if zero, cannot retry
* if positive, can retry after ≥N milliseconds.

Reply:
Under this condition, the client can decide time time duration of backing off by itself. This design made `use_backoff_time` and `backoff_time` as a hint, yep, this might be redundant.

And IMO, `retry_after` as a int(negative for cannot retry), or with a `can_retry` would be better, for it can cover the case "retry immediately".